tests/utils_tests/test_dateformat.py

```diff
@@ -142,9 +142,23 @@ def test_time_formats(self):
     def test_dateformat(self):
         my_birthday = datetime(1979, 7, 8, 22, 00)
 
-        self.assertEqual(dateformat.format(my_birthday, r"Y z \C\E\T"), "1979 189 CET")
-
-        self.assertEqual(dateformat.format(my_birthday, r"jS \o\f F"), "8th of July")
+        for format_, expected in [
+            # Simple case for escaping a word.
+            (r"jS \o\f F", "8th of July"),
+            # Simple cases with leading or trailing backslash.
+            # (Not using raw strings here as they cannot end with a backslash.)
+            ("\\jS \\o\\f F", "jth of July"),
+            ("jS \\o\\f F\\", "8th of July\\"),
+            # More pathological cases of nested escaping.
+            (r"Y z \C\E\T", "1979 189 CET"),
+            (r"Y z \\\C\\\E\\\T", r"1979 189 \C\E\T"),
+            (r"Y z \\\\\C\\\\\E\\\\\T", r"1979 189 \\C\\E\\T"),
+            # FIXME: These should work, but the current implementation is broken.
+            # (r"Y z \\C\\E\\T", r"1979 189 \C\July\CET"),
+            # (r"Y z \\\\C\\\\E\\\\T", r"1979 189 \\C\\July\\CET"),
```

Comment:
I would move these lines to PR with the appropriate fix. Do we have one?

Reply:
Sure, will do. This will be fixed as part of the main PR.

Reply:
Great, thanks!